### PR TITLE
Migrate newsletter/country/ page to post directly to Basket (Fixes #12376)

### LINKED ISSUE #12376: Newsletter /country page should post directly to Basket
## Description

URL: https://github.com/mozilla/bedrock/blob/main/bedrock/newsletter/urls.py#L25
Basket endpoint: https://github.com/mozmeao/basket/blob/master/basket/news/urls.py#L26

---

#### Success Criteria
- [x] Newsletter /country page posts directly to Basket using JS

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

- Updates `/newsletter/country/{TOKEN}/` page to post directly to Basket.
- Removes/redirects old `/newsletter/country/success/` URL as form now display success message using JS.

## Issue / Bugzilla link

#12376

## Testing

You need a newsletter token in order to view / test this page locally, which you can [get here](https://www.mozilla.org/en-US/newsletter/recovery/). Once you have a token you can view the page at:

`http://localhost:8000/en-US/newsletter/country/{TOKEN/`

### CODE REVIEW

File: media/js/newsletter/country.es6.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+import {
+    clearFormErrors,
+    disableFormFields,
+    enableFormFields,
+    postToBasket
+} from './form-utils.es6';
+
+let _form;
+let _countrySelect;
```
Comment (nathan-barrett) [inline media/js/newsletter/country.es6.js:15]: **Curious**: why use the underscore here?

File: tests/unit/spec/newsletter/country.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        </div>`;
+        document.body.insertAdjacentHTML('beforeend', form);
+    });
+
+    afterEach(function () {
+        const form = document.getElementById('country-newsletter-test-form');
+        form.parentNode.removeChild(form);
+    });
+
+    describe('form submission', function () {
```
Comment (nathan-barrett) [inline tests/unit/spec/newsletter/country.js:44]: Nice job on these test cases 🥳

**Review by nathan-barrett — APPROVED** [review]
r+ nice work on this alex!

File: media/js/newsletter/country.es6.js (hunk 1 above)
Comment (alexgibson) [reply media/js/newsletter/country.es6.js:15]: @nathan-barrett this is mostly a stylistic thing, to try and convey the variable in private to the ES module and is lexically scoped.

**Pull request merged**